### ISSUE TITLE
Warning when creating empty quasiquotes error

### ISSUE THREAD

**Issue by dten** [ndmitchell/hlint#766]
I'm trying to make this warning

```
  - warn: 
      lhs: "[hamlet| |]"
      rhs: "mempty"
```

but i get this error

```
hlint: user error (Failed to read YAML configuration file ~/project/.hlint.yaml
Aeson exception:
Error in $: Error when decoding YAML, Failed to parse Parse error: |], when parsing:
[hamlet| |]
Along path: root 7 warn lhs
When at: String
warn:
lhs: '[hamlet| |]'
rhs: mempty
)
```

i'm guessing these slices aren't supported, is there another way to achieve what i'm after :)?

**Comment by ndmitchell** [comment]
Does passing `-XQuasiQuotes` on the command line help? Do you normally pass `-XQuasiQuotes` to HLint in some way?

**Comment by dten** [comment]
it did not help

the arguments in the yaml are set to `  - arguments: [-XNoPatternSynonyms, -XNoRoleAnnotations, -XNumericUnderscores]` and nothing extra is passed on command line

**Comment by ndmitchell** [comment]
The issue is that QuasiQuotes isn't a default enabled extension. I wonder if I should enable even more extensions when parsing expressions in the .yaml files. They are highly unlikely to have problems with the kind of things that trip up syntax extensions (usually missing spaces) and it makes it easier to write things like you want.

**Comment by ndmitchell** [comment]
I just pushed a patch turning on more extensions when parsing config files - I think that should do the trick. If not, please let me know.

**Comment by dten** [comment]
Thanks :)

**Comment by dten** [comment]
just wanted to say this did work :) thanks

**Comment by dten** [comment]
sorry on further inspection it doesn't work! 🎢 

```
  - warn: {lhs: "[pizzapie| |]", rhs: "mempty", name: "Use mempty"}
```
i added the hint and indeed `[pizzapie| |]` does results in a warning. however... so does every quasi quote ever even with contents

**Comment by ndmitchell** [comment]
Thanks, I figured out the missing line, will send up a patch tomorrow.

**Comment by ndmitchell** [comment]
We weren't requiring literal strings in the AST nodes to match. I added that and now it works, plus a test.
